packages/framework/src/client/index.js

```diff
@@ -84,7 +94,13 @@ export default class ApiClient {
 	requireResource = ( requirement, resourceName, now = new Date() ) => {
 		const resources = this.state.resources || {};
 		const resourceState = resources[ resourceName ] || {};
-		this.scheduler.scheduleRequest( requirement, resourceState, resourceName, this.readOperationName, undefined, now );
+		// TODO: Remove this check after redux state is simplified out.
+		// This is necessary because components are getting re-rendered twice when dataReceived is dispatched.
+		// First before the state is updated, and second afterwards. This prevents resources from getting rescheduled
+		// on the first re-render. After redux dispatching is no longer used, components should no longer re-render twice.
+		if ( ! this.isStateStale ) {
```

Comment:
Minor nitpick in regards to variable names:

I feel like the request should be scheduled when the state is stale? I feel like when the state is "fresh" (i.e. not stale) there is no need to update, therefore no need to schedule?

Writing this out I see another contradiction here: why wouldn't you be able to schedule a request when the state is stale?

I understand that this is all stale vs. fresh in the context of this specific issue, but it feels like it can be easily be confused to represent something in the greater context of the system as a whole.
Am I reading too much into this (or misunderstanding the variable name?) or would it be better to choose a better name? Maybe `shouldScheduleRequest` or something along those lines?

Reply:
Yeah, maybe stale isn't correct and is overloaded in a module called "fresh-data" 😄 

Granted, this is temporary code, but even so I could change it to `isClientStateInSync` maybe, to show that the client object's state is either in sync or out of sync with the greater redux state?

Reply:
Yeah, that sounds better already! I might even go so far as to add a comment _somewhere_ explaining what this variable is _really_ about simply because I think it's hard to convey with a name 🤔 

I'll leave that up to you though, with the reminder that "temporary" code often turns out to be not-so-temporary 😃

Reply:
Resolved in 542c009